pkg/api/resource/quantity_proto.go

```diff
@@ -17,62 +17,287 @@ limitations under the License.
 package resource
 
 import (
-	"math/big"
+	"fmt"
+	"io"
 
-	inf "gopkg.in/inf.v0"
+	"github.com/gogo/protobuf/proto"
 )
 
-// QuantityProto is a struct that is equivalent to Quantity, but intended for
-// protobuf marshalling/unmarshalling. It is generated into a serialization
-// that matches Quantity. Do not use in Go structs.
-//
-// +protobuf=true
-type QuantityProto struct {
-	// The format of the quantity
-	Format Format `protobuf:"bytes,1,opt,name=format,casttype=Format"`
-	// The scale dimension of the value
-	Scale int32 `protobuf:"varint,2,opt,name=scale"`
-	// Bigint is serialized as a raw bytes array
-	Bigint []byte `protobuf:"bytes,3,opt,name=bigint"`
-}
+var _ proto.Sizer = &Quantity{}
 
-// ProtoTime returns the Time as a new ProtoTime value.
-func (q *Quantity) QuantityProto() *QuantityProto {
-	if q == nil {
-		return &QuantityProto{}
+func (m *Quantity) Marshal() (data []byte, err error) {
+	size := m.Size()
+	data = make([]byte, size)
+	n, err := m.MarshalTo(data)
+	if err != nil {
+		return nil, err
 	}
-	p := &QuantityProto{
-		Format: q.Format,
+	return data[:n], nil
+}
+
+// MarshalTo is a customized version of the generated Protobuf unmarshaler for a struct
+// with a single string field.
+func (m *Quantity) MarshalTo(data []byte) (int, error) {
+	var i int
+	_ = i
+	var l int
+	_ = l
+
+	data[i] = 0xa
+	i++
+	// BEGIN CUSTOM MARSHAL
+	out := m.toBytes()
+	i = encodeVarintGenerated(data, i, uint64(len(out)))
+	i += copy(data[i:], out)
+	// END CUSTOM MARSHAL
+
+	return i, nil
+}
+
+func encodeFixed64Generated(data []byte, offset int, v uint64) int {
+	data[offset] = uint8(v)
+	data[offset+1] = uint8(v >> 8)
+	data[offset+2] = uint8(v >> 16)
+	data[offset+3] = uint8(v >> 24)
+	data[offset+4] = uint8(v >> 32)
+	data[offset+5] = uint8(v >> 40)
+	data[offset+6] = uint8(v >> 48)
+	data[offset+7] = uint8(v >> 56)
+	return offset + 8
+}
+func encodeFixed32Generated(data []byte, offset int, v uint32) int {
+	data[offset] = uint8(v)
+	data[offset+1] = uint8(v >> 8)
+	data[offset+2] = uint8(v >> 16)
+	data[offset+3] = uint8(v >> 24)
+	return offset + 4
+}
+func encodeVarintGenerated(data []byte, offset int, v uint64) int {
+	for v >= 1<<7 {
+		data[offset] = uint8(v&0x7f | 0x80)
+		v >>= 7
+		offset++
 	}
-	if q.Amount != nil {
-		p.Scale = int32(q.Amount.Scale())
-		p.Bigint = q.Amount.UnscaledBig().Bytes()
+	data[offset] = uint8(v)
+	return offset + 1
+}
+func (m *Quantity) Size() (n int) {
+	var l int
+	_ = l
+
+	// BEGIN CUSTOM SIZE
+	l = len(m.toBytes())
+	// END CUSTOM SIZE
+
+	n += 1 + l + sovGenerated(uint64(l))
+	return n
+}
+
+func sovGenerated(x uint64) (n int) {
```

Comment:
maybe make this name reasonable, since it's unlikely we'll be rerunning the code generator?